evap/static/ts/src/sisyphus.ts

```diff
@@ -0,0 +1,309 @@
+                return false;
+            }
+            if (["BUTTON", "FIELDSET", "OBJECT", "OUTPUT"].includes(el.tagName)) {
+                return false;
+            }
+            return true;
+        });
+    }
+
+    getExcludeFields(): Element[] {
+        return this.options.excludeFields.flatMap(selector => Array.from(document.querySelectorAll(selector)));
+    }
+
+    getFormIdAndName(target: HTMLFormElement) {
+        return target.id + target.name;
+    }
+
+    getPrefix(target: HTMLFormElement, field: Element) {
+        return (
+            (this.options.locationBased ? this.href : "") +
+            this.getFormIdAndName(target) +
+            (field.getAttribute("name") || "") +
+            this.options.customKeySuffix
+        );
+    }
+
+    /**
+     * Bind saving data
+     */
+    bindSaveData() {
+        if (this.options.timeout) {
+            this.saveDataByTimeout();
+        }
+
+        for (const target of this.targets) {
+            for (const field of this.findFieldsToProtect(target)) {
+                if (this.getExcludeFields().includes(field)) {
+                    continue;
+                }
+                const prefix = this.getPrefix(target, field);
+                if (
+                    (field instanceof HTMLInputElement && field.type === "text") ||
+                    field instanceof HTMLTextAreaElement
+                ) {
+                    if (!this.options.timeout) {
+                        this.bindSaveDataImmediately(field, prefix);
+                    }
+                }
+                this.bindSaveDataOnChange(field);
+            }
+        }
+    }
+
+    /**
+     * Save all protected forms data to Local Storage.
+     * Common method, necessary to not lead astray user firing 'data is saved' when select/checkbox/radio
+     * is changed and saved, while text field data is saved only by timeout
+     */
+    saveAllData() {
+        for (const target of this.targets) {
+            let multiCheckboxCache: { [key: string]: boolean } = {};
+
+            for (const field of this.findFieldsToProtect(target)) {
+                if (this.getExcludeFields().includes(field) || !field.getAttribute("name")) {
+                    continue;
+                }
+                const prefix = this.getPrefix(target, field);
+                const fieldType = field.getAttribute("type");
+                // @ts-ignore
+                let value: string | string[] | boolean = field.value;
+
+                if (field instanceof HTMLInputElement && fieldType === "checkbox") {
+                    if (field.name.indexOf("[") !== -1) {
+                        if (multiCheckboxCache[field.name]) {
+                            return;
+                        }
+                        let tempValue: string[] = [];
+                        $("[name='" + field.name + "']:checked").each(function () {
+                            tempValue.push(field.value);
+                        });
```

Comment:
Re https://github.com/e-valuation/EvaP/pull/2377#pullrequestreview-2590630107: This is the only jquery left in here, right? Should be translatable to `querySelectorAll` -- would be really nice to have that